catkin_home/src/arm_vision/include/SegmentationNode/SegmentationNode.cpp

```diff
@@ -0,0 +1,83 @@
+#include "SegmentationNode.h"
+
+segmentation::segmentation(ros::NodeHandle nh, const std::string& camera_selection) : m_nh_(nh), camera_selection_(camera_selection) {
+    // define the subscriber and publisher
+    
+    if(camera_selection.compare("intel") == 0)
+    {
+        m_sub_ = m_nh_.subscribe("/camera/depth/color/points", 1, &segmentation::cloud_cb, this);
+        ROS_INFO("intel camera selected");
+
+    }
+    else if(camera_selection.compare("kinect") == 0)
+    {
+        m_sub_ = m_nh_.subscribe("/camera/depth_registered/points", 1, &segmentation::cloud_cb, this);
+        ROS_INFO("kinect camera selected");
+    }
+    
+    //m_sub_ = m_nh_.subscribe("/camera/depth/color/points", 1, &segmentation::cloud_cb, this);
+    m_cluster_pub_ = m_nh_.advertise<arm_vision::SegmentedClustersArray>("pcl_clusters", 1);
+    m_pub_ = m_nh_.advertise<sensor_msgs::PointCloud2>("output", 1);
+    m_pub2_ = m_nh_.advertise<sensor_msgs::PointCloud2>("output2", 1);
+}
+
+void segmentation::cloud_cb(const sensor_msgs::PointCloud2ConstPtr& input) {
+    // ... do data processing
+    sensor_msgs::PointCloud2 output;
+    pcl::PCLPointCloud2 outputPCL;
+
+    sensor_msgs::PointCloud2 output2;
+    pcl::PCLPointCloud2 outputPCL2;
+
+    // Container for original & filtered data
+    pcl::PCLPointCloud2::Ptr cloudorigin(new pcl::PCLPointCloud2);
+    pcl::PCLPointCloud2::Ptr cloud_filtered(new pcl::PCLPointCloud2);
+
+    // pcl::PCLPointCloud2ConstPtr cloudPtr(cloud);
+    // pcl::PCLPointCloud2 cloud_filtered;
+    // Convert to PCL data type
+    pcl_conversions::toPCL(*input, *cloudorigin);
+    // Perform the actual filtering
+    pcl::VoxelGrid<pcl::PCLPointCloud2> sor;
+    sor.setInputCloud(cloudorigin);
+    sor.setLeafSize(0.01, 0.01, 0.01);
```

Comment:
It is a good idea to keep constants defined somewhere else, to avoid ''magic numbers'